### PR TITLE
Add scheduler integration tests for creating objects out of order

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Exercise the multiple orders in which objects can be observed by kueue. They should all lead to admissible workloads.

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: test/integration/scheduler/scheduler_test.go [hunk 1]
```diff
@@ -596,6 +596,62 @@ var _ = ginkgo.Describe("Scheduler", func() {
 		})
 	})
 
+	ginkgo.When("Creating objects out-of-order", func() {
```
Comment (ahg-g) [inline test/integration/scheduler/scheduler_test.go:599]: can we also have a test for out-or-order deletion of resources

Comment (alculquicondor) [reply test/integration/scheduler/scheduler_test.go:599]: Are you asking to cover the finalizers? We have those in separate places.

Comment (ahg-g) [reply test/integration/scheduler/scheduler_test.go:599]: Yes, I was just thinking if we could have something analogous to this test; but maybe the ones we have are enough.

**Pull request merged**